x-pack/solutions/security/plugins/security_solution/public/common/utils/timeline/use_show_timeline_for_path.ts

```diff
@@ -9,30 +9,39 @@ import { useCallback, useMemo } from 'react';
 import { matchPath } from 'react-router-dom';
 
 import { getLinksWithHiddenTimeline } from '../../links';
-import { SourcererScopeName } from '../../../sourcerer/store/model';
-import { useSourcererDataView } from '../../../sourcerer/containers';
 import { useKibana } from '../../lib/kibana';
 import { hasAccessToSecuritySolution } from '../../../helpers_access';
 
+import { SourcererScopeName } from '../../../sourcerer/store/model';
+import { useSourcererDataView } from '../../../sourcerer/containers';
+import { useEnableExperimental } from '../../hooks/use_experimental_features';
+
 const isTimelinePathVisible = (currentPath: string): boolean => {
   const groupLinksWithHiddenTimelinePaths = getLinksWithHiddenTimeline().map((l) => l.path);
   const hiddenTimelineRoutes = groupLinksWithHiddenTimelinePaths;
   return !hiddenTimelineRoutes.find((route) => matchPath(currentPath, route));
 };
 
 export const useShowTimelineForGivenPath = () => {
-  const { indicesExist, dataViewId } = useSourcererDataView(SourcererScopeName.timeline);
   const {
     services: {
       application: { capabilities },
     },
   } = useKibana();
   const userHasSecuritySolutionVisible = hasAccessToSecuritySolution(capabilities);
 
-  const isTimelineAllowed = useMemo(
-    () => userHasSecuritySolutionVisible && (indicesExist || dataViewId === null),
-    [indicesExist, dataViewId, userHasSecuritySolutionVisible]
-  );
+  const { indicesExist, dataViewId } = useSourcererDataView(SourcererScopeName.timeline);
+
+  const { newDataViewPickerEnabled } = useEnableExperimental();
+
+  const isTimelineAllowed = useMemo(() => {
+    // NOTE: with new Data View Picker, data view is always defined
+    if (newDataViewPickerEnabled) {
+      return userHasSecuritySolutionVisible;
+    }
+
+    return userHasSecuritySolutionVisible && (indicesExist || dataViewId === null);
```

Comment:
I know this existed before, but why is `dataViewId === null` here? I would have expected it to be `!= null` ?